### PR TITLE
Broken limb debuffs

### LINKED ISSUE #1941: Broken leg is perfectly good for walking on
Having a broken leg does not impose any effect, and it should probably reduce the speed, weight limit and/or add pain when used for walking.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/1367062-broken-leg-is-perfectly-good-for-walking-on?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Ramza13**
#### Summary
```SUMMARY: Bugfixes "Added penalties for having broken limbs."```

#### Purpose of change
Fixes #1941 Broken leg is perfectly good for walking on, prevent wielding with two broken arms

#### Describe the solution
Prevent running with a broken leg and if you are on the ground(which broken legs and several other things cause) makes moving take 3 times as large.  

I have also added in a check to prevent wielding anything when you have two broken arms. That seems to me the minimal effect to be expected from two broken arms. Right now crafting is still allowed as long as you use the ground or a workbench. It might make sense to forbid that but then you can't craft splints so that's a no go.  Also fixed it so breaking an arm causes you to drop something you can no longer wield.

Last addition.  If you have two broken arms you can't use martial arts and your damage is now 10% of what it was.

Updated debug command to hurt self to allow choosing a body part.  A body part breaking from damage is different from one breaking from hp set to zero so this was awesome for testing.

#### Describe alternatives you've considered
The linked PR has people suggesting pain but realistically the problem with a broken leg isn't pain its that your leg doesn't work.  

I have also considered making the penalty extreme and adding crutches to help with it but that seems like it would have a lot of logic and additions and awkward checks for unbroken arms to use them and be larger than I currently want to try.

#### Testing
Broke legs and walked and attempted to run
Broke arms and attempted to wield one handed object.  With healthy arm wielded item, then broke arm.

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
As an alternative, we could restrict moving only to crouching with broken legs. So no walking and no running, just crawling.

File: src/player.h [hunk 1]
```diff
@@ -450,6 +450,7 @@ class player : public Character
         void cycle_move_mode(); // Cycles to the next move mode.
         void reset_move_mode(); // Resets to walking.
         void toggle_run_mode(); // Toggles running on/off.
+        bool can_run(); // source of truth of whether a player can run
```
Comment (KorGgenT) [inline src/player.h]: you should add this to Character, not player.

File: src/character.h [hunk 2]
```diff
@@ -459,12 +459,12 @@ class Character : public Creature, public visitable<Character>
...
         /** Returns the number of functioning legs */
         int get_working_leg_count() const;
-        /** Returns true if the limb is disabled */
+        /** Returns true if the limb is disabled(12.5% or less hp)*/
         bool is_limb_disabled( hp_part limb ) const;
-        /** Returns true if the limb is hindered(40% or less hp) */
-        bool is_limb_hindered( hp_part limb ) const;
         /** Returns true if the limb is broken */
         bool is_limb_broken( hp_part limb ) const;
+        /** source of truth of whether a player can run */
```
Comment (anothersimulacrum) [inline src/character.h]:
```suggestion
        /** source of truth of whether a Character can run */
```

Comment (anothersimulacrum) [reply src/character.h]: Or 
```suggestion
        /** Returns true if a player can run */
```

**Comment by Ramza13** [comment]
> 
> 
> As an alternative, we could restrict moving only to crouching with broken legs. So no walking and no running, just crawling.

This is growing on me. but I'm not sure crouching on its own is really slow enough to even pretend to be realistic for a broken limb or two.  

Actually that's another thing.  Right now one broken leg is the same penalty as two.  Does that make sense?

**Comment by Ramza13** [comment]
Ok so I have now added in a check to prevent wielding anything when you have two broken arms.  That seems to me the minimal effect to be expected from two broken arms.  Right now crafting is still allowed as long as you use the ground or a workbench.  It might make sense to forbid that but then you can't craft splints so that's a no go.

**Comment by Fris0uman** [comment]
Can you add a message like "You cannot run with two broken legs.", or something similar to tell players exactly what's wrong?

**Comment by Ramza13** [comment]
I did.  If you try it says you need two functional legs to run.

**Comment by Ramza13** [comment]
This should be the last addition.  If you are on the ground, which broken legs and other things cause or have two broken arms.  You can't use martial arts and your damage is now 10% of what it was.

File: src/melee.cpp [hunk 3]
```diff
@@ -486,6 +486,13 @@ void player::melee_attack( Creature &t, bool allow_special, const matec_id &forc
             technique_id = tec_none;
         }
 
+        // if you have two broken arms you aren't doing any martial arts
+        // and your hits are not going to hurt very much
+        if( get_working_arm_count() < 1 ) {
+            technique_id = tec_none;
+            d.mult_damage( .1 );
```
Comment (KorGgenT) [inline src/melee.cpp]:
```suggestion
            d.mult_damage( 0.1 );
```
to me this makes it more obvious that it's a decimal.

**Pull request merged**